pennylane/measurements/__init__.py

```diff
@@ -11,8 +11,151 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-This module contains measurements supported by PennyLane.
+r"""
+This module contains all the measurements supported by PennyLane.
+
+Description
+-----------
+
+Measurements
+~~~~~~~~~~~~
+The :class:`MeasurementProcess` class serves as a base class for measurements, and is inherited
+from the :class:`SampleMeasurement`, :class:`StateMeasurement` and :class:`MeasurementTransform`
+classes. These classes are subclassed to implement measurements in PennyLane.
+
+* Each :class:`SampleMeasurement` subclass represents a sample-based measurement, which contains a
+  :meth:`SampleMeasurement.process_samples` method that processes the sequence of samples generated
+  by the device. This method should always have the same arguments:
+
+  * samples (Sequence[complex]): computational basis samples generated for all wires
+  * wire_order (Wires): wires determining the subspace that ``samples`` acts on
+  * shot_range (tuple[int]): 2-tuple of integers specifying the range of samples
+        to use. If not specified, all samples are used.
+  * bin_size (int): Divides the shot range into bins of size ``bin_size``, and
+        returns the measurement statistic separately over each bin. If not
+        provided, the entire shot range is treated as a single bin.
+
+  `See :class:`CountsMP` for an example.
+
+* Each :class:`StateMeasurement` subclass represents a state-based measurement, which contains a
+  :meth:`StateMeasurement.process_state` method that processes the quantum state generated by the
+  device. This method should always have the same arguments:
+
+  * state (Sequence[complex]): quantum state
+  * wire_order (Wires): wires determining the subspace that ``state`` acts on; a matrix of
+        dimension :math:`2^n` acts on a subspace of :math:`n` wires
+
+  See :class:`StateMP` for an example.
+
+* Each :class:`MeasurementTransform` subclass represents a measurement process that requires
+  the application of a batch transform, which contains a :meth:`MeasurementTransform.process` method
+  that converts the given quantum script into a batch of quantum scripts and executes them using the
+  device. This method should always have the same arguments:
+
+  * qscript (QuantumScript): quantum script to transform
+  * device (Device): device used to transform the quantum script
+
+  The main difference between a :class:`MeasurementTransform` and a
+  :func:`~pennylane.batch_transform` is that a batch transform is tracked by the gradient transform,
+  while a :class:`MeasurementTransform` process isn't.
```

Comment:
Awesome! Is the `MeasurementTransform` a new feature for classical shadows?

Reply:
Yep!